lib/node_modules/@stdlib/blas/ext/index-of/lib/base.js

```diff
@@ -0,0 +1,109 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var dtypes = require( '@stdlib/ndarray/dtypes' );
+var gindexOf = require( '@stdlib/blas/ext/base/ndarray/gindex-of' );
+var dindexOf = require( '@stdlib/blas/ext/base/ndarray/dindex-of' );
+var sindexOf = require( '@stdlib/blas/ext/base/ndarray/sindex-of' );
+var factory = require( '@stdlib/ndarray/base/unary-reduce-strided1d-dispatch-factory' );
+
+
+// VARIABLES //
+
+var idtypes0 = dtypes( 'numeric_and_generic' ); // input ndarray
+var idtypes1 = dtypes( 'numeric_and_generic' ); // search element ndarray
+var idtypes2 = dtypes( 'integer_index_and_generic' ); // from index ndarray
+var odtypes = dtypes( 'integer_index_and_generic' );
+var policies = {
+	'output': 'integer_index_and_generic',
+	'casting': 'none'
+};
+var table = {
+	'types': [
+		'float64',
+		'float32'
+
+		// FIXME: add specialized support for `cindexOf` and `zindexOf` once the corresponding packages are implemented
+	],
+	'fcns': [
+		dindexOf,
+		sindexOf
+	],
+	'default': gindexOf
+};
+
+
+// MAIN //
+
+/**
+* Returns the first index of a specified search element along one or more ndarray dimensions.
+*
+* @private
+* @name indexOf
+* @type {Function}
+* @param {ndarrayLike} x - input ndarray
+* @param {(ndarrayLike|number)} searchElement - search element
+* @param {(ndarrayLike|number)} [fromIndex] - index from which to begin searching
+* @param {Options} [options] - function options
+* @param {IntegerArray} [options.dims] - list of dimensions over which to perform operation
+* @param {string} [options.dtype] - output ndarray data type
+* @throws {TypeError} first argument must be an ndarray-like object
+* @throws {TypeError} search element argument must be either an ndarray-like object or a scalar value
+* @throws {TypeError} from index argument must be either an ndarray-like object or a scalar value
```

Comment:
```suggestion
* @throws {TypeError} third argument must be either an ndarray-like object or a scalar value
```

Reply:
We try to avoid specifically naming arguments like you've done in order to avoid 1) copy-paste mistakes and 2) the potential for future drift when code is refactored, but docs are not. The exception we make is `options` which can "float" around to different argument positions depending on the API.